### ISSUE TITLE
Add type information to docstrings

### ISSUE THREAD

**Issue by ngreenwald** [angelolab/ark-analysis#141]
Currently we don't include information on argument types in a standardized way. We should adopt the google styleguide conventions and update our documentation as follows

```
def new_func(arg1, arg2):
      """ One sentence description
      Args:
            arg1 (str): description of arg1
            arg2 (int): description of arg2
     
      Returns:
            return1 (float): description of return1
"""
```

**Comment by ackagel** [comment]
Do we also want to start using type hints?
```
def new_func(arg1 : str, arg2 : int) -> float:
    ...
```
Could be overkill, but it sometimes makes error messages easier to diagnose.  Ex. `function, new_func, takes xarray but given string` versus `string has no attribute fovs`.

**Comment by ngreenwald** [comment]
Yeah, I wasn't sure. I don't think it looks as nice as having the types in
the docstring, but does give you the debugging benefit

**Comment by ackagel** [comment]
I think we can do without type hints for now.  I did a little more reading on them, and apparently they don't create runtime errors, only linter warnings, so they'd really be only for our own benefit/annoyance (not to mention it's annoying to manually define types for xarrays, dataframes, etc).  

tl;dr, I think doc-string types do basically the same thing for us, with 1/10th of the effort.

**Comment by ackagel** [comment]
after some pain I won't get into, I've found the functions which are lacking docstrings + types.

## Functions lacking formatted docstrings:
```
ark/utils/plot_utils.py:def plot_mod_ap(mod_ap_list, thresholds, labels):
ark/utils/plot_utils.py:def plot_error_types(errors, labels, error_plotting):
ark/utils/plot_utils.py:def barchart_helper(ax, values, labels, title):
```
As of today, the stitching functions are also docstring-less, but this is being resolved in PR #180

## Functions lacking types in their docstrings:
```
ark/analysis/visualize.py:def visualize_z_scores(z, pheno_titles):
ark/utils/plot_utils.py:def plot_color_map(outline_matrix, names, plotting_colors=None, ground_truth=None, save_path=None):
ark/utils/plot_utils.py:def plot_barchart_errors(pd_array, contour_errors, predicted_errors, save_path=None):
```

## Don't repeat this methodology
so that no one has to go down this rabbit hole again the commands to find these are:
```
ls ark/*/*.py | grep -v "test.py" | xargs -I {} cat {} | pcregrep -M -o1 '\t*(def .*?\(.+\):)\n+(\t|\s{4})[^"]' | xargs -I {} grep {} ark/*/*.py

ls ark/*/*.py | grep -v "test.py" | xargs -I {} cat {} | pcregrep -M -o1 '\t*(def .*?\(.+\):)\n+(\t|\s{4})"""[^"]*?\s{8}[^A-Z():]*?:[^"]*?"""' | xargs -I {} grep {} ark/*/*.py
```

# the moral of the story is that adam should've used python instead yep yep yep

**Comment by ngreenwald** [comment]
Thank you for your sacrifice. Let's hold off on the plot_utils for now, not
sure those are going to survive the winter.

**Comment by ngreenwald** [comment]
Closed by #196
